src/settings/environment.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+use std::env;
+
+use config::{Source, Value};
+
+pub trait QueryEnvironment {
```

Comment:
This trait wraps calls to the environment, which allows us to also implement a Mock environment for use in unit tests.

Reply:
This would make a really good inline comment; can we add it?